src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -0,0 +1,64 @@
+package org.broadinstitute.consent.http.service;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import org.broadinstitute.consent.http.db.DataSetDAO;
+
+import javax.inject.Inject;
+import java.util.Date;
+import org.broadinstitute.consent.http.models.DataSetProperty;
+import org.broadinstitute.consent.http.models.dto.DataSetDTO;
+
+public class DatasetService {
+
+    private final DataSetDAO dataSetDAO;
+
+    @Inject
+    public DatasetService(DataSetDAO dataSetDAO) {
+        this.dataSetDAO = dataSetDAO;
+    }
+
+    public DataSetDTO createDataset(String name, Set<DataSetProperty> properties) {
+        Date now = new Date();
+        int lastAlias = dataSetDAO.findLastAlias();
+        int alias = lastAlias + 1;
+
+        Integer id = dataSetDAO.insertDataset(name, now, null, false, alias);
+
+        List<DataSetProperty> propertyList = this.processDatasetProperties(id, now, properties);
+        dataSetDAO.insertDataSetsProperties(propertyList);
+
+        DataSetDTO result = new DataSetDTO();
+        Set<DataSetDTO> set = dataSetDAO.findDataSetWithPropertiesByDataSetIdWithOuterJoins(id);
+
+        for (DataSetDTO ds : set) {
+            result = ds;
+        }
+        return result;
+    }
+
+    // return -1 if no ds found
+    public Integer findDatasetByName(String name) {
```

Comment:
This should throw a `NotFoundException` if it can't find it. Callers of this method should understand that and handle it appropriately.